nucypher/blockchain/eth/interfaces.py

```diff
@@ -275,176 +265,94 @@ def _attach_provider(self, provider: Web3Providers = None, provider_uri: str = N
-            contract = self.client.w3.eth.contract(abi=interface['abi'],
-                                            bytecode=interface['bin'],
-                                            ContractFactoryClass=Contract)
-            return contract
-
-    def _wrap_contract(self,
-                       wrapper_contract: Contract,
-                       target_contract: Contract,
-                       factory=Contract) -> Contract:
-        """
-        Used for upgradeable contracts; Returns a new contract object assembled
-        with its own address but the abi of the other.
-        """
-
-        # Wrap the contract
-        wrapped_contract = self.client.w3.eth.contract(abi=target_contract.abi,
-                                                address=wrapper_contract.address,
-                                                ContractFactoryClass=factory)
-        return wrapped_contract
-
-    def get_contract_by_address(self, address: str):
-        """Read a single contract's data from the registrar and return it."""
-        try:
-            contract_records = self.registry.search(contract_address=address)
-        except RuntimeError:
-            # TODO #461: Integrate with Registry
-            raise self.InterfaceError(f'Corrupted contract registry: {self.registry.filepath}.')
-        else:
-            if not contract_records:
-                raise self.UnknownContract(f"No such contract with address: {address}.")
-            return contract_records[0]
 
-    def get_proxy(self, target_address: str, proxy_name: str, factory: Contract = Contract):
-
-        # Lookup proxies; Search for a registered proxy that targets this contract record
-        records = self.registry.search(contract_name=proxy_name)
+        #
+        # Confirm
+        #
 
-        dispatchers = list()
-        for name, addr, abi in records:
-            proxy_contract = self.client.w3.eth.contract(abi=abi,
-                                                  address=addr,
-                                                  ContractFactoryClass=factory)
+        # Primary check
+        deployment_status = receipt.get('status', UNKNOWN_TX_STATUS)
+        if deployment_status is 0:
+            failure = f"Transaction transmitted, but receipt returned status code 0. " \
+                      f"Full receipt: \n {pprint.pformat(receipt, indent=2)}"
+            raise self.InterfaceError(failure)
 
-            # Read this dispatchers target address from the blockchain
-            proxy_live_target_address = proxy_contract.functions.target().call()
+        if deployment_status is UNKNOWN_TX_STATUS:
+            self.log.info(f"Unknown transaction status for {txhash} (receipt did not contain a status field)")
 
-            if proxy_live_target_address == target_address:
-                dispatchers.append(proxy_contract)
+            # Secondary check TODO: Is this a sensible check?
+            tx = self.client.w3.eth.getTransaction(txhash)
```

Comment:
Should we add a `get_transaction` method to `Web3Client`?

Reply:
:heavy_check_mark: